UI/MapUI.gd

```diff
@@ -0,0 +1,104 @@
+var _MINIMUM_ROOM_WIDTH = 510
+var _MINIMUM_ROOM_HEIGHT = 490
+
+@export var color_rect: ColorRect
+@export var scroll_container: SmoothScrollContainer
+@export var room_container: ColorRect
+@export var room_addition_node: Control
+
+func _input(_inputevent: InputEvent) -> void:
+	if (_inputevent.is_action_pressed("cancel_action")):
+		queue_free()
+
+
+func _on_return_button_press():
+	queue_free()
+
+
+func _ready():
+	var current_map: MapBase = MapManager.current_map
+	
+	# Create New Room Object to append to the room container
+	var new_room: Control = room_ui.instantiate()
+	var new_room_texture_rect: TextureRect = Helpers.get_first_child_node_of_type(new_room, TextureRect)
+	var new_room_size: Vector2 = new_room_texture_rect.get_size()
+	
+	
+	var room_container_width: float = _get_combined_room_width(new_room_texture_rect)
+	# Set the max width between what we calculated above and the minimum room width constant
+	room_container_width = max(room_container_width, _MINIMUM_ROOM_WIDTH)
+	
+	var room_container_height = _get_combined_room_height(new_room_texture_rect)
+	# Set the max height between what we calculated above and the minimum room height constant
+	room_container_height = max(room_container_height, _MINIMUM_ROOM_HEIGHT)
+	
+	# Set the custom minimum size of the room container to allow scrolling
+	room_container.set_custom_minimum_size(Vector2(room_container_width, room_container_height))
+	
+	# Set the size of the scroll container to be dynamic to the max numbers of rooms of a floor
+	scroll_container.set_size(Vector2(room_container_width, scroll_container.get_size().y))
+	
+	# We're dynamically sizing the width of the map, as such we need to position it to center it on the screen.
+	# X position is calculated by getting half the width of the game screen, then subtracting that from 
+	# half the width of the scroll_container 
+	var scroll_container_position_x: float = color_rect.get_size().x / 2 - scroll_container.get_size().x / 2
+	scroll_container.set_position(Vector2(scroll_container_position_x, scroll_container.position.y))
+	
+	var scroll_container_bottom_y_position: float = scroll_container.position.y + scroll_container.get_size().y
+	
+	# Starting positions for the next room to be generated
+	# X = the Halfway position of room container - half the size of the longest floor to account for centering all of the rooms. 
+	#	  Add a little padding to push it away from the right edge.
+	# Y = Height of the Room Container that was calculated in new_room_container_height - (Screen Height - scroll container's bottom y position)
+	var start_position_for_next_room_x: float = room_container.position.x + _padding_offset
+	var start_position_for_next_room_y: float = room_container.get_custom_minimum_size().y - (color_rect.get_size().y - scroll_container_bottom_y_position)
+	var position_for_next_room: Vector2 = Vector2(start_position_for_next_room_x, start_position_for_next_room_y)
+	
+	for floor: Array[RoomBase] in current_map.rooms:
```

Comment:
godot throwing warning because `floor` is the name of a function, might be good to change this